am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/base/IntegrationBaseTest.java

```diff
@@ -8,13 +8,15 @@
 import org.junit.jupiter.api.BeforeEach;
 import org.testcontainers.containers.PostgreSQLContainer;
 import uk.gov.hmcts.reform.amlib.AccessManagementService;
+import uk.gov.hmcts.reform.amlib.DefaultRoleSetupImportService;
 
 @SuppressWarnings("PMD")
 public abstract class IntegrationBaseTest {
 
     private static final PostgreSQLContainer db = new PostgreSQLContainer().withUsername("sa").withPassword("");
     private static Jdbi jdbi;
     protected AccessManagementService ams;
+    protected DefaultRoleSetupImportService defaultRoleService;
```

Comment:
I think we should move both `AccessManagementService` and `DefaultRoleSetupImportService` to test subclasses so that each test class e.g. `FilterResourceIntegrationTest` should do something like:

```
private AccessManagementService service;

@BeforeAll
     void setup() {
         service = new AccessManagementService(
             db.getJdbcUrl(),
             db.getUsername(),
             db.getPassword()
         )
```

By the way, it makes sense to instantiate one service per test class, not one per test method as it is now. Service is stateless.

Adding more services to base test class does not make sense. It just unnecessarily adds more responsibilities to base class.

Reply:
and then each test subclass would have to define it's own instance of a PostregreSQLContainer?

Reply:
No, container can stay in `IntegrationBaseTest`. The intention of `IntegrationBaseTest` is to provide as access to DB. It should not really give us references to AM services.